r/src/array.cpp

```diff
@@ -119,4 +119,35 @@ std::shared_ptr<arrow::Array> DictionaryArray__dictionary(
   return array->dictionary();
 }
 
+// [[arrow::export]]
+std::shared_ptr<arrow::DataType> ListArray__value_type(
+    const std::shared_ptr<arrow::ListArray>& array) {
+  return array->value_type();
+}
+
+// [[arrow::export]]
+std::shared_ptr<arrow::Array> ListArray__values(
+    const std::shared_ptr<arrow::ListArray>& array) {
+  return array->values();
+}
+
+// [[arrow::export]]
+int32_t ListArray__value_length(const std::shared_ptr<arrow::ListArray>& array,
```

Comment:
These are "low-level" functions, they don't perform any bound checking, you can segfault with the wrong `i`. Do they need to be exported?

Reply:
🤷‍♂ what's the rule about what is export worthy ?